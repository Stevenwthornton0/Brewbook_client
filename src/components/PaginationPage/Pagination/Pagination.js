import React, { Component } from 'react';
import './Pagination.css'

class Pagination extends Component {

    static defaultProps = {
        handlePagination: () => {}
    }
    
    render() {
        const { page, currentPage } = this.props

        if (page === currentPage) {
            return ( <button className='disabled' disabled={true}>{page}</button> )
        }

        return (
            <button onClick={() => {this.props.handlePagination(page)}}>{page}</button>
        )
    }
}

export default Pagination;

