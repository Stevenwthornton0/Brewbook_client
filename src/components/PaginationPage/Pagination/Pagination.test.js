import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import Pagination from './Pagination';

Enzyme.configure({ adapter: new Adapter() })

describe(`Pagination smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<Pagination />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})