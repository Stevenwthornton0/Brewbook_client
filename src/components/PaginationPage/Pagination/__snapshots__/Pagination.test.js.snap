// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination smoke test renders without crashing 1`] = `
<button
  className="disabled"
  disabled={true}
/>
`;
